macsec: rename parent property to base-iface

ffmancera opened the pull request:
Rename the parent property to base-iface to be consistent with other interfaces like VLAN or VXLAN.

cathay4t commented:
@ffmancera Please fix the CI failure on `test_gen_conf_for_examples`

liangwen12year reviewed (commented):
CI failed on "libnmstate.error.NmstateValueError: Invalid YAML string: missing field `base-iface`", please take a look.

ffmancera committed:
macsec: rename parent property to base-iface

Rename the parent property to base-iface to be consistent with other
interfaces like VLAN or VXLAN.

Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>
Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
I have fixed the patch and CI is all passing now.
